### PR TITLE
[Playwright] new e2e test for atcp self-consent enrollment

### PR DESCRIPTION
**Pull request by aweng98**
[PEPPER-966](https://broadworkbench.atlassian.net/browse/PEPPER-966)

Self-consent participantion in ATCP study.

All tests have passed on [CircleCI](https://app.circleci.com/pipelines/github/broadinstitute/ddp-angular/25218/workflows/5bd704cf-8aca-47bd-a1c8-db4dc29925ab).

[PEPPER-966]: https://broadworkbench.atlassian.net/browse/PEPPER-966?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -707,7 +707,7 @@ jobs:
             TESTS_FILE=$(circleci tests split --split-by=timings --timings-type=classname e2e_tests.txt)
             echo $TESTS_FILE
             npm run test:ci $TESTS_FILE
-          no_output_timeout: 3m # Default is 10m. Set shorter time to ensure CI fails faster if test hangs
+          no_output_timeout: 5m # Default is 10m. Set shorter time to ensure CI fails faster if test hangs
```
Comment (aweng98) [inline .circleci/config.yml:710]: Longer time neeed as more tests is making backend server runs slower.

File: playwright-e2e/dss/component/input.ts [hunk 2]
```diff
@@ -37,8 +38,12 @@ export default class Input extends WidgetBase {
    * @param opts
    * @returns {Promise<void>}
    */
-  async fill(value: string | number, opts?: { dropdownOption: string, type?: boolean }): Promise<void> {
-    const useType = opts?.type ? opts.type : false;
+  async fill(value: string | number, opts: { dropdownOption?: string, type?: boolean, nth?: number, waitForSaveRequest?: boolean } = {}): Promise<void> {
```
Comment (aweng98) [inline playwright-e2e/dss/component/input.ts:41]: Add new parameter to include waiting of `/answers` request as some input fields have this request automatically triggered when out of focus.

File: playwright-e2e/dss/component/table.ts [hunk 3]
```diff
@@ -108,23 +108,41 @@ export default class Table {
...
    */
   async getHeaderNames(): Promise<Array<string>> {
     return this.headerLocator().allInnerTexts();
-    /*
-    const columns = await this.headerLocator().elementHandles();
-    return await Promise.all(
-      _.map(columns, async (column) => {
-        return await column.innerText();
-      })
-    ); */
```
Comment (aweng98) [inline playwright-e2e/dss/component/table.ts:127]: No longer need to keep.

File: playwright-e2e/utils/file-utils.ts [hunk 4]
```diff
@@ -8,8 +8,9 @@ export interface MailListCSV {
   lastName?: string;
 }
 
-export function createTextFileSync(pathName: string, data: string) {
+export function createTextFileSync(dir: string, pathName: string, data: string) {
   try {
+    fs.mkdirSync(dir, { recursive: true });
```
Comment (aweng98) [inline playwright-e2e/utils/file-utils.ts:13]: Resolve DSM Kits Upload tests failures with this error.
```
saliva-kit-upload-flow.spec.ts:48:9 Should upload a single kit for one participant @functional @visual @dsm @Leiomyosarcoma
[chromium] › tests/dsm/kitUploadFlow/saliva-kit-upload-flow.spec.ts:48:9 › Saliva Kits upload flow › Should upload a single kit for one participant @functional @visual @dsm @Leiomyosarcoma 

    Error: ENOENT: no such file or directory, open '/home/circleci/repo/playwright-e2e/test-results/tests-dsm-kitUploadFlow-saliva-kit-upload-flow-411bc-ant-functional-visual-dsm-Leiomyosarcoma-08e8b1-chromium/SALIVA_Leiomyosarcoma-1689284669007.txt'

       at utils/file-utils.ts:13

      11 | export function createTextFileSync(pathName: string, data: string) {
      12 |   try {
    > 13 |     fs.writeFileSync(pathName, data);
         |        ^
      14 |     console.log(`File: ${pathName} - created successfully`)
      15 |   } catch (error) {
      16 |     console.error(`Couldn't create the File: ${pathName}`);

        at createTextFileSync (/home/circleci/repo/playwright-e2e/utils/file-utils.ts:13:8)
        at KitUploadPage.uploadFile (/home/circleci/repo/playwright-e2e/dsm/pages/kitUpload-page/kitUpload-page.ts:32:23)
        at /home/circleci/repo/playwright-e2e/tests/dsm/kitUploadFlow/saliva-kit-upload-flow.spec.ts:134:27

    attachment #1: screenshot (image/png) ──────────────────────────────────────────────────────────
    test-results/tests-dsm-kitUploadFlow-saliva-kit-upload-flow-411bc-ant-functional-visual-dsm-Leiomyosarcoma-08e8b1-chromium/test-failed-1.png
```

File: playwright-e2e/dsm/pages/participant-list-page.ts [hunk 5]
```diff
@@ -207,4 +207,44 @@ export default class ParticipantListPage {
     await button.click();
     await waitForNoSpinner(this.page);
   }
+
+  async findParticipantForKitUpload(): Promise<number> {
```
Comment (aweng98) [inline playwright-e2e/dsm/pages/participant-list-page.ts:211]: Took code from DSM Kits Upload tests to remove duplicated code.

**Review by Rinchan20 — APPROVED** [review]

**Pull request merged**